src/librustc_mir/build/matches/test.rs

```diff
@@ -756,6 +818,26 @@ impl<'a, 'gcx, 'tcx> Builder<'a, 'gcx, 'tcx> {
     }
 }
 
+impl Test<'_> {
+    pub(super) fn targets(&self) -> usize {
+        match self.kind {
+            TestKind::Eq { .. } | TestKind::Range(_) | TestKind::Len { .. } => {
+                2
+            }
+            TestKind::Switch { adt_def, .. } => {
+                adt_def.variants.len() + 1
+            }
+            TestKind::SwitchInt { switch_ty, ref options, .. } => {
+                if switch_ty.is_bool() {
```

Comment:
I'm trying to figure out the overall picture here.

My current best guess is that, for `SwitchInt` on a `bool`, the `options` ends up holding just some permutation of `{ 0, 1 }` (and we know that to be exhaustive, since the type is `bool`), and thus `options.len() + 1` would give the wrong number of target basic blocks.

Regardless of whether my hypothesis above is correct or incorrect, it might be nice to add some docs to `SwitchInt` explaining the role of the `options` field, and what invariants it upholds with respect to the other fields (`switch_ty` and `indices`).